### DISCUSSION TITLE
MDF/MF4 file formats

### DISCUSSION THREAD

**Discussion by SonnyOvens** [collin80/SavvyCAN#680] [Ideas]
Do you think there will ever be support for directly opening MDF or MF4 files without first converting to a text based format? These are the files produced by the CANedge devices made by CSS Electronics (and others).

**Comment by collin80** [comment]
Maybe. I'm looking into it. I already support loading of some Canalyzer binary files that are pretty similar in terms of features. In that case I really only do the bare minimum to load CAN frames and don't support much else. MF4 may end up similarly. That is, I doubt that writing will be coming soon but it's possible. In general I try to support reading as many formats as possible. So, we'll see where this goes.

**Comment by SonnyOvens** [comment]
That would be awesome. We wouldn’t have much use for writing, but reading would be great!!On Oct 9, 2023, at 17:07, Collin Kidder ***@***.***> wrote:﻿
Maybe. I'm looking into it. I already support loading of some Canalyzer binary files that are pretty similar in terms of features. In that case I really only do the bare minimum to load CAN frames and don't support much else. MF4 may end up similarly. That is, I doubt that writing will be coming soon but it's possible. In general I try to support reading as many formats as possible. So, we'll see where this goes.

**Comment by ihedvall** [comment]
@SonnyOvens @collin80 
I have a C++ library that both read and writes ASAM MDF files (GitHub::ihedvall/mdflib). It have an MIT license. It support the ASAM MDF bus logger standard. 

Ingemar Hedvall
